### PR TITLE
Add unannotated_bg_ratio parameter to pre_filtering function

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

Recent decisions have changed the data filtering flow.
It was argued that filtering without using unannotated items as background might not be suitable for some training, 
so making this configurable.

- Add `unannotated_bg_ratio` in `otx.core.config.data.DataModuleConfig`
- Sampling based on ratio (Default: 0.0) first, then filter.

How to use
`otx train --config src/otx/recipe/detection/atss_mobilenetv2.yaml --data_root detection/vitens_large --data.config.unannotated_items_ratio 0.1`

```
original dataset length: 2833
empty_item (train): 652
unannotated_items_ratio: 0.1
-> filtered dataset length: 2246 (2833 - (652 - int(652*0.1)))
```

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/recipe/_base_/data/mmaction_base.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ config:
     - 500
     - 500
   image_color_channel: RGB
+  unannotated_bg_ratio: 0.0
```
Comment (vinnamkim) [inline src/otx/recipe/_base_/data/mmaction_base.yaml]:
```suggestion
  unannotated_items_ratio: 0.0
```
I'm feeling that "unannotated" and "bg" are semantically duplicated each other.

File: src/otx/core/data/pre_filtering.py [hunk 2]
```diff
@@ -15,21 +16,35 @@
...
+    Raises:
+        None.
+    """
+    used_background_items = []
+    msg = f"There are empty annotation items in train set, Of these, only {unannotated_bg_ratio*100}% are used."
     warnings.warn(msg, stacklevel=2)
-    return False
+    if unannotated_bg_ratio > 0:
+        empty_items = [item for item in dataset if item.subset == "train" and len(item.annotations) == 0]
+        used_background_items = sample(empty_items, int(len(empty_items) * unannotated_bg_ratio))
```
Comment (vinnamkim) [inline src/otx/core/data/pre_filtering.py]:
```suggestion
    if unannotated_bg_ratio > 0:
        empty_item_ids = [item.id for item in dataset if item.subset == "train" and len(item.annotations) == 0]
        used_background_ids = set(sample(empty_items, int(len(empty_items) * unannotated_bg_ratio)))
```
1. Checking `DatasetItem` equivalence is heavier than `str`'s.
2. `item in list` is a O(n) algorithm.

Comment (harimkang) [reply src/otx/core/data/pre_filtering.py]: Thanks for letting me know what I missed. - https://github.com/openvinotoolkit/training_extensions/pull/2965/commits/ba350f2b7cbedf68e7306874b7e02a7ec36a95e4

File: src/otx/recipe/_base_/data/mmaction_base.yaml (hunk 1 above)
Comment (harimkang) [reply src/otx/recipe/_base_/data/mmaction_base.yaml]: Thank you for the suggestion. - https://github.com/openvinotoolkit/training_extensions/pull/2965/commits/ba350f2b7cbedf68e7306874b7e02a7ec36a95e4

**Review by jaegukhyun — APPROVED** [review]

**Review by vinnamkim — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Pull request merged**